docs(state-query): add zustand to newdocs (#1912)
latticexyz/mud: next-docs/pages/state-query/typescript/zustand.mdx

import { CollapseCode } from "../../../components/CollapseCode";

# Zustand

[Zustand](https://docs.pmnd.rs/zustand/getting-started/introduction) is a state management library that supports JavaScript and TypeScript.
You can see it in use in the [`react` template](https://github.com/latticexyz/mud/tree/main/templates/react).

> ## Under React

Reviewer: ```suggestion
## With React
```

> Zustand is React native.

Reviewer: ```suggestion
Zustand is native to React.
```

Reviewer: (to avoid confusion with "React Native")

> The way you use it is very similar to the way you'd use `useState`, except that the state you receive is bigger, it is auto-populated by MUD, and you change it through a `System` instead of using a function provided by the hook.

Reviewer: ```suggestion
The way you use it is very similar to the way you'd use `useState`, but with a bigger state tree that is synchronised with the contract state by MUD. To change state you send a transaction to a [System](/world/systems).
```
